### PR TITLE
Deprecating hpx/traits.hpp

### PR DESCRIPTION
**Pull request by hkaiser**
- make sure that traits usage has proper headers

### CODE REVIEW

**Comment by sithhell** [comment]
Can we try to consistent about the order of includes?
I think most files follow the approach to include hpx stuff first, then other dependencies and last the stdlib includes. Within those groups I think we mostly order the list of includes alphabetically. Since this PR is mostly shuffling around headers, it would be a nice starting point for a consistent policy.

**Comment by hkaiser** [comment]
> Can we try to consistent about the order of includes?

Absolutely!

> I think most files follow the approach to include hpx stuff first, then other dependencies and last the stdlib includes. 

That's what we consistently do, already.

> Within those groups I think we mostly order the list of includes alphabetically. 

Can we use a tool for that?

> Since this PR is mostly shuffling around headers, it would be a nice starting point for a consistent policy.

Hmmm, the main intention for this PR was to get rid of `hpx/traits.hpp`. Most 'header shuffling' was done to make sure the traits definitions where properly included when used.

File: hpx/traits/wrap_int.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_TRAITS_WRAP_INT_JUN_03_2016_0835PM)
+#define HPX_TRAITS_WRAP_INT_JUN_03_2016_0835PM
+
+#include <hpx/config.hpp>
+
+namespace hpx { namespace traits { namespace detail
```
Comment (K-ballo) [inline hpx/traits/wrap_int.hpp]: This file should be in `traits/detail`.

**Comment by hkaiser** [comment]
As a test, I used clang-format with the following `.clang-format` file to sort the #includes in all *.hpp files:

```
---
Language:        Cpp
DisableFormat:   true
IncludeCategories:
  - Regex:           '^<hpx/config\.hpp>'
    Priority:        1
  - Regex:           '^<hpx/.*\.hpp>'
    Priority:        2
  - Regex:           '^<hpx/parallel/.*\.hpp>'
    Priority:        3
  - Regex:           '^<.*'
    Priority:        4
  - Regex:           '.*'
    Priority:        5
SortIncludes:    true
...
```

**Comment by sithhell** [comment]
Thanks! LGTM. I guess we should really work on getting a common .clang-format up and running...

**Comment by hkaiser** [comment]
> I guess we should really work on getting a common .clang-format up and running...

I tried that. I was not really happy with what's possible.

**Pull request merged**